compiler/noirc_frontend/src/elaborator/path_resolution.rs

```diff
@@ -107,7 +107,10 @@ impl<'context> Elaborator<'context> {
     pub(super) fn resolve_path(&mut self, mut path: Path) -> PathResolutionResult {
         let mut module_id = self.module_id();
 
-        if path.kind == PathKind::Plain && path.first_name() == SELF_TYPE_NAME {
+        if path.kind == PathKind::Plain
+            && !path.segments.is_empty()
+            && path.first_name() == SELF_TYPE_NAME
```

Comment:
Maybe `first_name` should default to `""` if the segments are empty?
That way this'd likely fix any other uses of `first_name` as well.

Reply:
I initially made `first_name` return `Option<&str>` but wasn't sure... would you prefer that instead?

Reply:
I'd prefer defaulting to "" or the Option version over panicking internally if there are cases where we don't actually have any path segments

Reply:
Done!